Storage/Service/AbstractUrlStorage.php

```diff
@@ -2,193 +2,89 @@
 
 namespace EMS\CommonBundle\Storage\Service;
 
+use GuzzleHttp\Psr7\Stream;
 use Psr\Http\Message\StreamInterface;
-use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 
 abstract class AbstractUrlStorage implements StorageInterface
 {
 
-    /**
-     * Returns the base url of the storage service
-     * @return string
-     */
-    abstract protected function getBaseUrl(): string;
+   abstract protected function getBaseUrl(): string;
 
-    /**
-     * returns the a file path or a resource url that can be handled by file function such as fopen
-     */
-    protected function getPath(string $hash, ?string $cacheContext = null, bool $confirmed = true, string $ds = '/'): string
+    protected function getPath(string $hash, bool $confirmed = true, string $ds = '/'): string
     {
-        $folderName = $this->getBaseUrl();
+        $folderName = $this->getBaseUrl() . $ds . $confirmed ? substr($hash, 0, 3) : 'uploads';
```

Comment:
I would not put an if statement into a string concatenation. More difficult to read.

Reply:
I see that $confirmed is set to false for initUpload(). Instead of passing this bool I would split up into "getPath()" and "getUploadPath" to use "upload"?

Reply:
See 0515b6fb938a3b24bba576b840d6938ad7976f1c